xtraplatform-feature-provider-sql/src/main/java/de/ii/xtraplatform/feature/provider/sql/app/ExtentReaderSql.java

```diff
@@ -61,4 +61,39 @@
         return LogContextStream.graphWithMdc(sourceStream.map(sqlRow -> sqlDialect.parseExtent((String) sqlRow.getValues()
                                                                                                               .get(0), crs)), Sink.head());
     }
+
+    public RunnableGraphWithMdc<CompletionStage<Optional<Period>>> getTemporalExtent(FeatureStoreTypeInfo typeInfo, String property) {
+        FeatureStoreInstanceContainer instanceContainer = typeInfo.getInstanceContainers()
+                .get(0);
+        Optional<FeatureStoreAttributesContainer> temporalAttributesContainer = instanceContainer.getTemporalAttributesContainer(property);
+
+        if (!temporalAttributesContainer.isPresent()) {
+            throw new IllegalArgumentException("temporal property not found:" + property);
+        }
+
+        String query = queryGenerator.getTemporalExtentQuery(temporalAttributesContainer.get(), property);
+
+        Source<SqlRow, NotUsed> sourceStream = sqlConnector.getSqlClient().getSourceStream(query, SqlQueryOptions.withColumnTypes(String.class));
+
+        return LogContextStream.graphWithMdc(sourceStream.map(sqlRow -> sqlDialect.parseTemporalExtent((String) sqlRow.getValues().get(0))), Sink.head());
```

Comment:
The method SqlDialect.parseTemporalExtent() doesn't exist yet, should it be added?

Reply:
Yes, that makes sense. As the SQL query should have two response columns, the method should have two parameters:

`sqlDialect.parseTemporalExtent((String) sqlRow.getValues().get(0), (String) sqlRow.getValues().get(1))`

The method would then create the `Interval` using `Instant.parse` and `Interval.of`.